packages/boot/src/__tests__/integration/model.booter.integration.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright IBM Corp. 2019,2020. All Rights Reserved.
+// Node module: @loopback/boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect, TestSandbox} from '@loopback/testlab';
+import {resolve} from 'path';
+import {BooterApp} from '../fixtures/application';
+
+describe('repository booter integration tests', () => {
+  const sandbox = new TestSandbox(resolve(__dirname, '../../.sandbox'));
+
+  const MODELS_TAG = 'model';
+
+  let app: BooterApp;
+
+  beforeEach('reset sandbox', () => sandbox.reset());
+  beforeEach(getApp);
+
+  it('boots repositories when app.boot() is called', async () => {
+    const expectedBindings = [
+      'models.Model1',
+      'models.Model2',
+      'models.NoEntity',
+      'models.Product',
+    ];
+
+    await app.boot();
+
+    const bindings = app.findByTag(MODELS_TAG).map(b => b.key);
+    expect(bindings.sort()).to.eql(expectedBindings.sort());
+  });
+
+  async function getApp() {
+    await sandbox.copyFile(resolve(__dirname, '../fixtures/application.js'));
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/no-entity.model.js'),
+      'models/no-entity.model.js',
+    );
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/product.model.js'),
+      'models/product.model.js',
+    );
+
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/multiple-models.model.js'),
+      'models/multiple-models.model.js',
+    );
+
+    const MyApp = require(resolve(sandbox.path, 'application.js')).BooterApp;
+    app = new MyApp();
+  }
```

Comment:
Shouldn't the fixtures' extension be `.ts`?

```suggestion
  async function getApp() {
    await sandbox.copyFile(resolve(__dirname, '../fixtures/application.ts'));
    await sandbox.copyFile(
      resolve(__dirname, '../fixtures/no-entity.model.ts'),
      'models/no-entity.model.ts',
    );
    await sandbox.copyFile(
      resolve(__dirname, '../fixtures/product.model.ts'),
      'models/product.model.ts',
    );

    await sandbox.copyFile(
      resolve(__dirname, '../fixtures/multiple-models.model.ts'),
      'models/multiple-models.model.ts',
    );

    const MyApp = require(resolve(sandbox.path, 'application.ts')).BooterApp;
    app = new MyApp();
  }
```

Reply:
@achrinza We always only scan the transpiled `.js` code as the booters need to load them.

Reply:
@raymondfeng Ah, gotcha. Slipped my mind that it'll already be transpiled, my bad 🙇